### PR TITLE
Convert client to TS part 2

### PR DESCRIPTION
**Pull request by albertogasparin**
Another batch of components in `desktop-client` converted to TS

### CODE REVIEW

File: packages/desktop-client/src/components/manager/subscribe/ConfirmPasswordForm.tsx [hunk 1]
```diff
@@ -44,14 +44,18 @@ export function ConfirmPasswordForm({ buttons, onSetPassword, onError }) {
...
+          setPassword1(e.target.value)
+        }
         onEnter={onSubmit}
       />
       <Input
         placeholder="Confirm password"
         type={showPassword ? 'text' : 'password'}
         value={password2}
-        onChange={e => setPassword2(e.target.value)}
+        onChange={(e: ChangeEvent<HTMLInputElement>) =>
```
[the remark this answers is not in the document]
Comment (albertogasparin) [reply packages/desktop-client/src/components/manager/subscribe/ConfirmPasswordForm.tsx:56]: Could not find a way 😞 . TS and React do not play nice with event types. There are a couple of other places in this PR where you will see I had to cast to a more specific type to stop TS complaining

File: .eslintrc.js [hunk 2]
```diff
@@ -37,8 +37,6 @@ module.exports = {
       require('confusing-browser-globals').filter(g => g !== 'self'),
     ),
 
-    'react/jsx-no-useless-fragment': 'error',
```
[the remark this answers is not in the document]
Comment (albertogasparin) [reply .eslintrc.js:40]: Had to remove this rule because flags false positives (where we need fragments to make TS happy, eg wrapping an array of nodes). Overall adding fragments has no implications anyway

[the remark this answers is not in the document]
Comment (j-f1) [reply .eslintrc.js:40]: Can you switch to per-line disables then? That way we catch any excess fragments in the future. (They can be annoying since they force their contents to be indented). Ideally we wouldn’t have to add the fragments too often anyway.

File: packages/desktop-client/src/components/budget/rollover/TransferTooltip.tsx [hunk 3]
```diff
@@ -69,7 +82,7 @@ export default function TransferTooltip({
         <InitialFocus>
           <Input
             value={amount}
-            onChange={e => setAmount(e.target.value)}
+            onChange={e => setAmount(e.target['value'])}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/rollover/TransferTooltip.tsx:85]: **❓ question:** ‏out of curiosity - why did you change this here and in a bunch of other places?

**Review by MatissJanis — APPROVED** [review]
Sweet!

File: packages/desktop-client/src/components/budget/rollover/TransferTooltip.tsx (hunk 3 above)
Comment (albertogasparin) [reply packages/desktop-client/src/components/budget/rollover/TransferTooltip.tsx:85]: This is a way to tell Typescript that the property we are trying to get exists on the object. In this case `e.target` might or might not have `value` and we could either do `(e.target as HTMLInputElement).value` or "force" it via `e.target['value']`. Dunno if using `currentTarget` would [solve the problem properly](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12239) but that would have been a runtime change I didn't wanna add in this PR.
